packages/core/contracts/UnirepSocial.sol

```diff
@@ -265,4 +269,39 @@ contract UnirepSocial {
     ) external {
         unirep.updateUserStateRoot(publicSignals, proof);
     }
+
+     /*
+     * Set new user name for an epochKey
+     * @param epochKey epoch key that attempts to set a new uername
+     * @param oldUsername oldusername that the eppch key previously claimed
+     * @param newUsername requested new user name
+     */
+     function setUsername(
+        uint256 epochKey,
```

Comment:
I am not sure whether `epochKey` should be verified on-chain so that it will not be easily changed by other users (if we don't want an `admin` to control)